test/generators/html/Stop_dead_link_doc.html

```diff
@@ -112,25 +112,87 @@ <h1>Module <code><span>Stop_dead_link_doc</span></code></h1>
+      <span> = </span>
+     </code>
+     <ol>
+      <li id="type-another_foo.Bar" class="def variant constructor anchored">
+       <a href="#type-another_foo.Bar" class="anchor"></a>
+       <code><span>| </span>
+        <span><span class="constructor">Bar</span> 
+         <span class="keyword">of</span> 
+         <span class="xref-unresolved">{Another_Foo}1.t</span>
+        </span>
+       </code>
+      </li>
+     </ol>
     </div>
    </div>
    <div class="odoc-spec">
     <div class="spec type anchored" id="type-another_bar">
      <a href="#type-another_bar" class="anchor"></a>
-     <code><span><span class="keyword">type</span> another_bar</span></code>
+     <code><span><span class="keyword">type</span> another_bar</span>
+      <span> = </span>
+     </code>
+     <ol>
+      <li id="type-another_bar.Bar" class="def variant constructor anchored">
+       <a href="#type-another_bar.Bar" class="anchor"></a>
+       <code><span>| </span>
+        <span><span class="constructor">Bar</span> 
+         <span class="keyword">of</span> 
+        </span><span>{</span>
+       </code>
+       <ol>
+        <li id="type-another_bar.field" class="def record field anchored">
+         <a href="#type-another_bar.field" class="anchor"></a>
+         <code>
+          <span>field : <span class="xref-unresolved">{Another_Foo}1.t</span>
+           ;
+          </span>
+         </code>
+        </li>
+       </ol><code><span>}</span></code>
+      </li>
+     </ol>
     </div>
    </div>
    <div class="odoc-spec">
     <div class="spec type anchored" id="type-another_foo_">
      <a href="#type-another_foo_" class="anchor"></a>
-     <code><span><span class="keyword">type</span> another_foo_</span></code>
+     <code><span><span class="keyword">type</span> another_foo_</span>
+      <span> = </span>
+     </code>
+     <ol>
+      <li id="type-another_foo_.Bar_" class="def variant constructor
+       anchored"><a href="#type-another_foo_.Bar_" class="anchor"></a>
+       <code><span>| </span>
+        <span><span class="constructor">Bar_</span> 
+         <span class="keyword">of</span> int * 
+         <span class="xref-unresolved">{Another_Foo}1.t</span> * int
+        </span>
+       </code>
+      </li>
+     </ol>
     </div>
    </div>
    <div class="odoc-spec">
     <div class="spec type anchored" id="type-another_bar_">
      <a href="#type-another_bar_" class="anchor"></a>
-     <code><span><span class="keyword">type</span> another_bar_</span></code>
+     <code><span><span class="keyword">type</span> another_bar_</span>
+      <span> = </span>
+     </code>
+     <ol>
+      <li id="type-another_bar_.Bar__" class="def variant constructor
+       anchored"><a href="#type-another_bar_.Bar__" class="anchor"></a>
+       <code><span>| </span>
+        <span><span class="constructor">Bar__</span> 
+         <span class="keyword">of</span> 
+         <span><span class="xref-unresolved">{Another_Foo}1.t</span> option
```

Comment:
I think internal idents shouldn't be outputted. The useful parts of this type declaration are: Exhaustive list of the constructors, arity of each constructors.

I suggest rendering types containing hidden paths as `_`. This is not valid OCaml but neither is the proposed output.

This example could render as `Bar of _` if we keep this specific to variants or `Bar of _ option` if we extend this to every type expressions.

Reply:
Yeah I'm quite sympathetic to this point of view, but I'd rather address this in a larger scoped PR, as this is a much bigger issue than in just the variants.

Reply:
In the meantime, we could have this for variants only ? Eg. render `Bar of _`.

Reply:
That would require special-casing the rendering of paths when they're found in variants, so I don't want to go down this route at this point.